feat: Create PL for several SOs and Create multiple DN from PL, grouped by customer.

KrithiRamani opened the pull request:
1. To enable creation of PL for SOs of several customers, removed validation for Customer field.

<img width="1080" alt="Screenshot 2021-09-20 at 11 57 24 AM" src="https://user-images.githubusercontent.com/12999179/134944865-cbb9e229-2da4-42ce-b480-b18dddbbb3fb.png">

2. Set Picked Qty to zero, on select of Items. On submit, the Picked Qty will be set to Stock Qty.
<img width="1080" alt="Screenshot 2021-09-20 at 12 01 30 PM" src="https://user-images.githubusercontent.com/12999179/134944989-0df7696b-234b-40cc-aa4f-f68cc95b02d7.png">

3. From a PL created from several SOs, on click of Create -> Delivery Note
       - Separate DN for sales orders(grouped by customer) will be created
       - Separate DN for all items that do not have any SO linked.
<img width="1440" alt="Screenshot 2021-09-27 at 9 45 18 PM" src="https://user-images.githubusercontent.com/12999179/134946505-89d81fdf-ffb1-4b82-a260-1b1a851c8e61.png">
<img width="1440" alt="Screenshot 2021-09-27 at 9 45 38 PM" src="https://user-images.githubusercontent.com/12999179/134946542-efe357d9-b916-4c0d-8456-570a86befec9.png">

4. Added new field per_picked in SO, set this on submit of PL. Used this field to filter SO in Get Items of PL.
<img width="1440" alt="Screenshot 2021-09-27 at 9 47 53 PM" src="https://user-images.githubusercontent.com/12999179/134946840-d2c71b6c-69fe-4f39-bc49-7f9bee22954b.png">

18alantom committed:
test: check execution of illegal stock entry seq

rohitwaghchaure commented on erpnext/stock/doctype/pick_list/pick_list.js:
```diff
@@ -143,13 +143,10 @@ frappe.ui.form.on('Pick List', {
 			docstatus: 1,
 			per_delivered: ['<', 100],
 			status: ['!=', ''],
+			per_picked: ['=',0],
```
per_picked < 100, could be partial

rohitwaghchaure commented on erpnext/stock/doctype/pick_list/pick_list.py:
```diff
@@ -41,6 +51,24 @@ def before_submit(self):
+	def update_so(self):
+		for item in self.locations:
+			if item.sales_order_item:
+				frappe.db.set_value("Sales Order Item",item.sales_order_item,"picked_qty",item.picked_qty)
```
What if same item picked multiple times in the different pick lists? For example in the sales order item's qty is 5 and use has made 2 pick list with 4 qty in the first pick list and 1 qty in the second pick list

KrithiRamani replied:
Added validation on submit of PL.

rohitwaghchaure commented on erpnext/stock/doctype/pick_list/pick_list.py:
```diff
@@ -41,6 +51,24 @@ def before_submit(self):
+				frappe.db.set_value("Sales Order Item",item.sales_order_item,"picked_qty",item.picked_qty)
+				so_doc = frappe.get_doc("Sales Order",item.sales_order)
+				if self.docstatus == 2:
+					per_picked = 0
```
What if there is one more pick list exists

rohitwaghchaure commented on erpnext/stock/doctype/pick_list/pick_list.py:
```diff
@@ -27,9 +27,19 @@ def validate(self):
 	def before_submit(self):
 		for item in self.locations:
+			# set picked_qty to stock_qty, before submit
+			item.picked_qty = item.stock_qty
```
if item.picked_qty is set by the user manually then shouldn't be override

KrithiRamani replied:
Handled.

rohitwaghchaure commented on erpnext/stock/doctype/pick_list/pick_list.py:
```diff
@@ -314,64 +344,88 @@ def get_available_item_locations_for_other_item(item_code, from_warehouses, requ
+		delivery_note.insert(ignore_mandatory = True)
 
+	frappe.msgprint('Delivery Note(s) created for the Pick List!')
 	return delivery_note
```
create_delivery_note looks a big function so difficult to read , could you split it into smaller functions

KrithiRamani replied:
Done

deepeshgarg007 committed:
Merge pull request #28385 from vorasmit/fix-hsn_wise

fix(regional): `hsn_wise` as false returns `item_code`

deepeshgarg007 committed:
fix: Move trigger from on trash to on cancel

deepeshgarg007 committed:
Merge pull request #28464 from mhbu50/develop

fix:Change QR Code Triggerr event

GangaManoj committed:
Merge branch 'develop' into serialized-item-consumption

nemesis189 committed:
feat(pos): Total item qty field to POS screen (#28331)

nextchamp-saqib committed:
feat: Let serialised Items be consumed during Asset Repairs (#28349)

deepeshgarg007 committed:
fix: Paid showing in AR/AP report

(cherry picked from commit 5e7ce5370f6af634f7674772529cf4933114f3ef)

ruchamahabal committed:
fix: allow creating Shift Assignment for same day (#28613)

ankush committed:
fix(ux): remove attachment limits

deepeshgarg007 committed:
feat: Show Zero Values filter in consolidated financial statement

ruchamahabal committed:
fix: Employee Transfer and Project Profitability test cases (#28633)

* fix: Employee Transfer testcases

* fix: Project Profitability test case

barredterra committed:
feat: Grant commission on certain items only (#27467)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

ankush committed:
chore: remove duplicate code (#28646)

[skip ci]

deepeshgarg007 committed:
Merge pull request #28636 from deepeshgarg007/zero_values_report

feat: Show Zero Values filter in consolidated financial statement

mergify[bot] committed:
fix: Unable to search project by project name in Sales Invoice (bp #28648) 

(cherry picked from commit 08b7c856b2ee94d1f8ac2c019c556eef4d7dd7da)

Co-authored-by: Deepesh Garg <42651287+deepeshgarg007@users.noreply.github.com>

deepeshgarg007 committed:
fix: Taxes and Charges template not getting copied from Purchase Order/Receipt to Invoice

(cherry picked from commit 6a75e8d283bb76214af832bb0a29c20eefae328c)

nemesis189 committed:
fix: Taxjar Nexus list visible only if child table is visible

GangaManoj committed:
fix: Add item to packing list

sagarvora committed:
fix: update modified timestamp for delivery note #28657

fix: update `modified` timestamp for delivery note

nemesis189 committed:
fix: removing db call for variables

nextchamp-saqib committed:
fix: cannot load company form (#28535)

rtdany10 committed:
fix(POS Profile): replace `cur_frm` with `frm` (#28390)

nextchamp-saqib committed:
Merge branch 'develop' into taxjar-nexus-display-condition

ankush committed:
test: timeout certain tests in work order to avoid stuck tests (#28666)

[skip ci]

ankush committed:
test: dynamic fiscal year creation in tests (#28667)

GangaManoj committed:
fix: Fix depreciation_amount calculation

GangaManoj committed:
fix: Create Depreciation Schedules properly for existing Assets

GangaManoj committed:
fix: Modify has_pro_rata() to include existing assets

GangaManoj committed:
fix: Test if depreciation schedules are set up properly for existing assets

GangaManoj committed:
Merge branch 'develop' into product-bundle-fixes

GangaManoj committed:
fix: Remove unnecessary variable

GangaManoj committed:
fix: Fix 'Adjust Asset Value' button

GangaManoj committed:
fix: Rename variable

GangaManoj committed:
fix: Rename function

deepeshgarg007 committed:
Merge pull request #28533 from GangaManoj/product-bundle-fixes

fix: misc minor Product Bundle fixes

nextchamp-saqib committed:
Merge branch 'develop' into depreciate-existing-asset

pateljannat committed:
fix: new item form tour for home onboarding

pateljannat committed:
fix: assigned proper form tour to onboarding steps

GangaManoj committed:
fix: Make buttons translatable (#28679)

nextchamp-saqib committed:
Merge branch 'develop' into taxjar-nexus-display-condition

nextchamp-saqib committed:
fix: actual tax conversion in case of multicurrency invoices (#28539)

nextchamp-saqib committed:
Merge branch 'develop' into depreciate-existing-asset

nextchamp-saqib committed:
refactor(SLA): remove response_by_variance & resolution_by_variance

nextchamp-saqib committed:
fix: failing tests

ankush committed:
fix: dont requeue repost immediately and clear progress (#28684)

nextchamp-saqib committed:
fix: failing tests

deepeshgarg007 committed:
fix: India utils code cleanup

(cherry picked from commit 56c626adbfbd04e7e9063f6500ec380f7bfb3da4)

deepeshgarg007 committed:
Merge pull request #28693 from frappe/mergify/bp/develop/pr-28627

fix: Paid invoices showing in AR/AP report (backport #28627)

nextchamp-saqib committed:
fix: qrcode image name for invoices with special chars (#28699)

nextchamp-saqib committed:
fix: Create Depreciation Schedules for existing Assets accurately (#28675)

nextchamp-saqib committed:
fix: failing tests

ruchamahabal committed:
fix(Non Profit): fetch memberships for 80G certificate by from date only (#28700)

ankush committed:
refactor: replace misleading variable name

nextchamp-saqib committed:
Merge branch 'develop' into fix-sla

ankush committed:
refactor: simplify the way SLEs are submitted

noahjacob committed:
fix: incorrect outgoing rates when material_consumption enabled

deepeshgarg007 committed:
Merge pull request #28698 from frappe/mergify/bp/develop/pr-28654

fix: Taxes and Charges template not getting copied from Purchase Order/Receipt to Invoice (backport #28654)

deepeshgarg007 committed:
fix: Invocie amount in KSA E Invoice QR Code

deepeshgarg007 committed:
Merge pull request #28708 from deepeshgarg007/ksa_qr_invoice_total

fix: Invoice amount in KSA E Invoice QR Code

ankush committed:
fix!: dont allow renaming warehouse primary key

ankush committed:
fix: remove autocommit from item rename

nextchamp-saqib committed:
fix(ksa): qrcode for invoices with special chars (#28715)

nextchamp-saqib committed:
fix: undo removing of resolution_time fields

noahjacob committed:
test: added tests for manufacture stock entry when material_consumption is enabled

ankush committed:
fix: weird item sorting by `idx`

rohitwaghchaure committed:
Merge pull request #28710 from noahjacob/calc_outgoing_stock_entry

fix: incorrect outgoing rates when material_consumption enabled

deepeshgarg007 committed:
fix: Better Error logging fordeferred revenue/expense booking

deepeshgarg007 committed:
fix: Commit joural entries

deepeshgarg007 committed:
fix: Log error before throwing exception

ruchamahabal committed:
fix(patch): create only component type field instead of running the whole setup (#28734)

ruchamahabal committed:
feat: Exit Interview

ruchamahabal committed:
feat: sending Exit Questionnaire

ruchamahabal committed:
feat: add default Exit Questionnaire email template

ruchamahabal committed:
feat: track status and final decision (Retained/Exit Confirmed)

ruchamahabal committed:
fix: make Exit Interview submittable

ruchamahabal committed:
feat: bulk questionnaire sending

ruchamahabal committed:
feat: update Exit Interview date in employee master on submission

ankush committed:
fix: remove bad defaults from selling settings

"All cusotmer groups" and "All territories" are pointless defaults, not
sure why these are made default. They don't help you track anything.
"All" might as well be `Null`.

Even the filters for customer_group suggest it shouldn't be group then
having the root as default makes no sense.

ruchamahabal committed:
feat: Employee Exits report

ankush committed:
fix: patch to remove default item group and territory

ruchamahabal committed:
chore: update HR workspace

ankush committed:
test: set customer group and territory defaults

ruchamahabal committed:
feat: default Notification - a day before Exit Interview

ruchamahabal committed:
fix: email summary

nextchamp-saqib committed:
fix(test): test_depr_entry_posting_with_income_account

nextchamp-saqib committed:
feat: Make Depreciation Entry posting more flexible (#28421)

deepeshgarg007 committed:
Merge pull request #28656 from nemesis189/taxjar-nexus-display-condition

fix: Taxjar Nexus list visible only if child table is visible

noahjacob committed:
refactor: map serial from schedule if only one

nextchamp-saqib committed:
patch: split 'ongoing' sla status

anupamvs committed:
feat: CRM Settings (#27788)

* feat: crm settings

* feat: CRM Settings

* feat: lead and opprtunity section

* feat: added CRM Settings in ERPNext Settings workspace

* fix: review chnages

* added patch

* fix: linter issues

* fix: linter issues

* fix: linter issues

* fix: removed crm settings from selling module

* fix: raw query to frappe.qb

* fix: removed hardcoded value

* fix: linter issue

* fix: simplify CRM Settings migration patch

Co-authored-by: Anupam Kumar <anupam@Anupams-MacBook-Air.local>
Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

ankush committed:
Merge pull request #28745 from noahjacob/maint_batch_map

refactor: map serial from schedule if only one

nextchamp-saqib committed:
refactor: handle special cases on communication creation

nextchamp-saqib committed:
feat: record assignment on first response failure

nextchamp-saqib committed:
Merge branch 'develop' into fix-sla

deepeshgarg007 committed:
fix: Ageing in AR/AP report for advances

nextchamp-saqib committed:
fix: transalations

nextchamp-saqib committed:
fix: sider issues

nextchamp-saqib committed:
fix: test

nextchamp-saqib committed:
fix: time to respond & resolve indicators

nextchamp-saqib committed:
fix: remove missed 'ongoing' references

nextchamp-saqib committed:
fix: remove 'ongoing' status from issue summary report

nextchamp-saqib committed:
fix: indentation

deepeshgarg007 committed:
fix: TDS Monthly payable report

rtdany10 committed:
feat: create QR Code field if not existing

rtdany10 committed:
fix: get doctype from doc instead of hardcoding SI

rtdany10 committed:
fix: localize QR fieldname

rtdany10 committed:
chore: change QR field label

rtdany10 committed:
feat: create QR field in POS Invoice

ruchamahabal committed:
fix: show Exit Questionnaire button only to the users with write access

- fix linter issues

rtdany10 committed:
fix: rename KSA QR field to match localisation

ruchamahabal committed:
test: Exit Interview

ruchamahabal committed:
chore: fix report column widths

ankush committed:
test: index test should pass without calling function #28771

test: index test should pass without calling function

[skip ci]

rtdany10 committed:
fix: reload SI meta before checking

rtdany10 committed:
Merge branch 'frappe:develop' into ksa-vat-updates

deepeshgarg007 committed:
Merge pull request #28731 from deepeshgarg007/better_deferred_revenue_error_logging

fix: Better Error logging for deferred revenue/expense booking

ruchamahabal committed:
test: Employee Exits Report

ruchamahabal committed:
Merge branch 'develop' into exit-interview

ankush committed:
fix: ignore mandatory fields while creating WO from SO (#28772)

If fields are made mandatory from customizations the WO creation simply
fails.

rtdany10 committed:
feat: generate QR Code for POS transactions

rtdany10 committed:
fix: disable ksa vat invoice by default

rtdany10 committed:
feat(Regional): patch to disable ksa print formats for other countries

rtdany10 committed:
feat(Regional): POS Invoice format for KSA

rtdany10 committed:
feat(Regional): enable KSA print formats on setup

rtdany10 committed:
fix: reload doc

rtdany10 committed:
Merge branch 'frappe:develop' into ksa-vat-updates

rtdany10 committed:
feat(Regional): rename qr field and disable print formats

rtdany10 committed:
chore: linters

rtdany10 committed:
fix: intendation error

deepeshgarg007 committed:
fix: Error on creating invoice

(cherry picked from commit e11515a3561eac6d33d21190cac2db112ae301fb)

deepeshgarg007 committed:
Merge pull request #28779 from frappe/mergify/bp/develop/pr-28777

fix: Error on creating invoice (backport #28777)

ankush committed:
fix: check future negative stock for batches

batch's ledger is only maintained in form of `actual_qty` on batch's
SLEs. To validate if batch has any negative qty in future, cumulative
total of `actual_qty` is required to ensure it never goes negative.

ankush committed:
refactor: remove redundant batch qty validation

This check was only checking total sum, which is problamatic when making
backdated entries that can cause intermediate values to go negative
while overall values stay positive.

ruchamahabal committed:
fix: tests and sider issues

rtdany10 committed:
chore: switch to frappe.db.set_value

rtdany10 committed:
chore: switch to ORM method

rtdany10 committed:
fix: replace frappe.get_all

deepeshgarg007 committed:
Merge branch 'develop' into ksa-vat-updates

nextchamp-saqib committed:
fix: hsn-wise summary is incorrect if an invoice has repeated item code

ankush committed:
test: simplfy test and expect specific exception

rtdany10 committed:
chore: POS invoice format changes

ankush committed:
test: add multi-batch negative qty test

deepeshgarg007 committed:
fix: Error on Invoice generation

(cherry picked from commit 82255293c4eb8a9f586db083d19d63c1ea435fb9)

deepeshgarg007 committed:
Merge pull request #28790 from frappe/mergify/bp/develop/pr-28786

fix: Error on Invoice generation (backport #28786)

aaronmenezes committed:
fix: Maintenence Visit -Purpose (item ) tables is not visible on submitted or saved entries (#28792)

GangaManoj committed:
Merge branch 'develop' into fix-clear-depr-schedule

GangaManoj committed:
Merge branch 'develop' into fix-adjust-asset-value

deepeshgarg007 committed:
Merge pull request #28776 from rtdany10/ksa-vat-updates

feat(Regional): KSA E-Invoing optimizations and POS support

michael-koeller committed:
fix: wrong german translation of abbreviation PAN

Wrong german translation of abbreviation: PAN

ankush committed:
fix: deduplicate after finishing the repost (#28803)

Not really a bug but avoids potential of prematurely skipping something
if failure occurs and failure isn't resolved.

s-aga-r committed:
fix: misleading "Set Default X" fields after saving (#28798)

* fix: misleading "Set Default X" fields after saving

* refactor: remove unncessary code and minor formatting

* fix: extend to more doctypes and correct fieldnames

Co-authored-by: Ankush Menat <ankush@frappe.io>

noahjacob committed:
feat: added QI link in Job Card Dashboard (#28643)

ankush committed:
fix: ensure that reposting is finished before freezing stock/account

ankush committed:
test: stock frozen validation

ankush committed:
chore: update stale rules and apply on issues

[skip ci]

ankush committed:
chore: new and improved bug report form

[skip ci]

rohitwaghchaure committed:
Merge pull request #28815 from ankush/frozen_repost

fix: validate pending reposts before freezing stock/account

ankush committed:
chore: correct form format for issues

ankush committed:
chore: remove unused issue tempaltes

[skip ci]

ruthra-kumar committed:
feat: Deferred Revenue and Expense report

 - show deferred revenue and expense with actual and expected postings
 - unit tests added

hrwX committed:
feat(buying): quotation number in supplier quotation (#28420)

deepeshgarg007 committed:
fix: KSA QR field rename patch

deepeshgarg007 committed:
Merge pull request #28764 from deepeshgarg007/tds_report_fixes

fix: TDS Monthly payable report

deepeshgarg007 committed:
Merge pull request #28748 from deepeshgarg007/accounts_receivable_jv_due_date

fix: Ageing in AR/AP report for advances

nextchamp-saqib committed:
test: test_hsn_summary_for_invoice_with_duplicate_items

ruchamahabal committed:
Merge branch 'develop' into exit-interview

ruchamahabal committed:
fix: missing import

ruchamahabal committed:
fix: tests

- specify sorting order in employee exits query

- rollback after work order tests

ruchamahabal committed:
feat(HR): Exit Interview and Employee Exits Report (#28741)

* feat: Exit Interview

* feat: sending Exit Questionnaire

* feat: add default Exit Questionnaire email template

* feat: track status and final decision (Retained/Exit Confirmed)

* fix: make Exit Interview submittable

* feat: bulk questionnaire sending

* feat: update Exit Interview date in employee master on submission

* feat: Employee Exits report

* chore: update HR workspace

* feat: default Notification - a day before Exit Interview

* fix: email summary

* fix: show Exit Questionnaire button only to the users with write access

- fix linter issues

* test: Exit Interview

* chore: fix report column widths

* test: Employee Exits Report

* fix: tests and sider issues

* fix: missing import

* fix: tests

- specify sorting order in employee exits query

- rollback after work order tests

nextchamp-saqib committed:
Merge pull request #28550 from nextchamp-saqib/fix-sla

refactor(SLA): Application of SLA and its fields

ankush committed:
Merge pull request #28839 from ankush/migration_hashes

fix: remove migration hashes from doctypes

ruthra-kumar committed:
feat: new column 'Time taken to Deliver' in sales order analysis

pateljannat committed:
Merge branch 'develop' into home-onboarding

nextchamp-saqib committed:
fix: validate if asset account is set against company

nextchamp-saqib committed:
fix: Clear Depreciation Schedule before modification (#28507)

nextchamp-saqib committed:
Merge pull request #28677 from GangaManoj/fix-adjust-asset-value

fix: Fix 'Adjust Asset Value' button

nextchamp-saqib committed:
revert: "fix: expected value after useful life validation"

nextchamp-saqib committed:
Merge pull request #28783 from nextchamp-saqib/hsn-wise-summary-report-fix

fix: hsn-wise summary is incorrect if an invoice has repeated item code

ankush committed:
fix: packing slip incorrect item value (#28862)

noahjacob committed:
feat: added required_date field to set date in child table (#28432)

* feat: added required_date field to set date in child table

* fix: dont refresh full table

Co-authored-by: Ankush Menat <ankush@frappe.io>

noahjacob committed:
fix(UX): hide RM table(Job Card) if material transfer is against work order (#28746)

deepeshgarg007 committed:
Merge pull request #28832 from deepeshgarg007/ksa_e_invoice_patch_fix

fix: KSA QR field rename patch

ankush committed:
fix: reorder patch to fix custom field options (#28866)

ruchamahabal committed:
Merge pull request #28845 from ruchamahabal/fix-salary-slip-timesheet

fix: incorrect amount based on payment days in timesheet salary slip

aaronmenezes committed:
fix: Nonstock item showing in Itemwise Recommended Reorder Level report (#28873)

ruthra-kumar committed:
refactor: remove unused function 'get_postings'

ankush committed:
fix: incorrect logic for  "Reserved Qty for Production" (#28880)

* fix: reservation for production incorrect

The query uses case to decide what fields to compute reservation on,
this case is outermost case hence the very first Work order's "Skip
transfer" is considered for ALL work orders.

Solution: move the case inside Sum.

Steps to reproduce:
1. Make work order for more than 1 qty (with | without skip transfer)
2. Create manufacture and transfer entries.
3. Keep checking reserved quantities during this process.

* test: use default warehouse for testing reservation

ankush committed:
fix: remove set only once from valuation method (#28876)

ankush committed:
test: remove commits from tests (#27939)

* test: remove commits from tests in selling module

* test: remove commits from manufacturing module

* test: remove commits from accounts module tests

* test: remove commits from controller tests

* test: keep commits in accounting dimension test

* Revert "test: remove commits from controller tests"

This reverts commit f5cfbee5da844f8b2bffeee907ac602754f16ff4.

* Revert "test: remove commits from accounts module tests"

This reverts commit df4c27efddf355e2731a18e8871e54d18590f219.

nextchamp-saqib committed:
fix: reset_expected_response_and_resolution on hold

ankush committed:
chore: bump coverage requirements

ruthra-kumar committed:
refactor: change field to duration and fetch elapsed seconds

nextchamp-saqib committed:
fix: failing tests

ruthra-kumar committed:
refactor: rename checkbox and translations

 - rename checkbox and make it
 - added translation text for labels
 - fixes for linter error

ruchamahabal committed:
fix(test): Employee Exit Summary (#28888)

nextchamp-saqib committed:
Merge pull request #28887 from nextchamp-saqib/sla-fixes-#1

fix: reset_expected_response_and_resolution on hold

nextchamp-saqib committed:
revert: "fix: expected value after useful life validation"

nextchamp-saqib committed:
Merge branch 'develop' into so-days-taken-to-deliver

nextchamp-saqib committed:
Merge branch 'develop' into home-onboarding

nextchamp-saqib committed:
fix: time to response shows failed (#28895)

nemesis189 committed:
fix: Currency fix for `cost` field in subscription plan (#28821)

nemesis189 committed:
feat: 'Invoice Number' field in Opening Invoice Creation Tool (#28844)

deepeshgarg007 committed:
Merge branch 'develop' into deferred_report_revenue_and_expense

pateljannat committed:
fix: titles and order

pateljannat committed:
Merge branch 'home-onboarding' of https://github.com/pateljannat/erpnext into home-onboarding

pateljannat committed:
Merge branch 'develop' into home-onboarding

deepeshgarg007 committed:
Merge pull request #28822 from ruthra-kumar/deferred_report_revenue_and_expense

feat: Deferred Revenue and Expense report with actual and upcoming postings

FHenry committed:
feat: Manufacturing : in BOM Operation time can be fix (#27063)

* feat: Operation time can be fixed

* better coding without duplicate code

* change modified date foir doctype

* undo change

* fix varmixing

* fix varmixing

* fix: remove f-string that's not required anymore

* refactor: use ORM instead of raw query

* fix: improve description

* test: basic test for fixed time

Co-authored-by: Ankush Menat <ankush@frappe.io>

nextchamp-saqib committed:
feat: Home Onboarding (#28551)

ankush committed:
perf: remove forcing of posting sort index on stock balance (#28902)

mysql is generally smart enough to figure out which index is better
based on cardinality of index. While posting sort index is better for
low item high # of SLE scenario it's bad for high item variety each with
lower count of SLEs.

DeeMysterio committed:
feat: provision to have limited parameters for in-process quality ins… (#28549)

* feat: provision to have limited parameters for in-process quality inspection

* chore: formatting

nextchamp-saqib committed:
fix(ksa): enable ksa pos invoice print format

deepeshgarg007 committed:
fix: Multicurrency invoices using subscription

deepeshgarg007 committed:
fix: Add tests for multicurrency subscription

ankush committed:
chore: ledger invariant check report (#28921)

noahjacob committed:
fix: mapping to maintenance visit gets erased (#28917)

* fix: mappings to maintenance visit gets erased

* test: mapping of schedule to visit

deepeshgarg007 committed:
Merge pull request #28916 from deepeshgarg007/multicurrency_subscription

fix: Multicurrency invoices using subscription

deepeshgarg007 committed:
Merge pull request #28911 from nextchamp-saqib/fix-ksa-pf

fix(ksa): enable ksa pos invoice print format

mergify[bot] committed:
fix: convert asynchronous field update to synchronous (#28930)

(cherry picked from commit 5adf85f5b7ba94591381b03709b1fee6aa9331a5)

Co-authored-by: Dan Navarro on Ubuntu Work <danjeremynavarro@gmail.com>

nemesis189 committed:
fix: Add Invoice Number field to list view in Opening Inv Creation Tool (#28938)

rtdany10 committed:
fix: remove rounding of avg rating (#28945)

ankush committed:
refactor: move Fifo Valuation to a new class

ankush committed:
feat: repr, eq and iter for fifo queue

ankush committed:
test: FifoValuation tests

ankush committed:
fix: round off if near zero while popping from queue

ankush committed:
test: hypothesis tests for FIFO queue

ankush committed:
feat: get consumed bins from Fifo queue on pop

ankush committed:
refactor: outgoing rate and defaults for remove_stock

ankush committed:
fix: return consumed bin on collapsing queue

ankush committed:
perf: specify slots on FIFO queue

ankush committed:
chore: rename FifoValuation to FIFOValuation

ankush committed:
fix: correct bin qty on backdated transactions

When making a backdated transactions current balance qty depends on
evaluation of whole ledger inbetween, instead of doing that just fetch
the last sle's qty_after_transaction when future transactions are
detected against SLE

fix: don't update bin's actual_qty

1. it's already updated by repost_current_voucher
2. update if future sle exists

ankush committed:
test: bin qty on backdated SR

deepeshgarg007 committed:
fix: Error on viewing consolidated financial statement

ankush committed:
chore: drop deprecated qunit tests

deepeshgarg007 committed:
Merge pull request #28959 from deepeshgarg007/consolidated_balance_return_fix

fix: Error on viewing consolidated financial statement

anupamvs committed:
Merge pull request #28440 from anupamvs/crm-lead-validate-cleanup-develop

fix: lead validate function cleanup

ankush committed:
test: remove unnecessary creation of companies #28965

test: remove unnecessary creation of companies

ruchamahabal committed:
fix: fetch appointment letter content in the same order as template (#28968)

nemesis189 committed:
fix: Added filter for dispatch address (#28937)

nextchamp-saqib committed:
Merge branch 'develop' into so-days-taken-to-deliver

nextchamp-saqib committed:
feat: new column 'Time taken to Deliver' in sales order analysis (#28842)

GangaManoj committed:
fix: Reset value_after_depreciation on reversing journal entry during Asset return

developmentforpeople committed:
fix: missed colon (#28979)

ankush committed:
style: better labels for SLE fields (#28988)

* style: better labels for SLE fields

* style: rename stock queue field

[skip ci]

Co-Authored-by: marination <maricadsouza221197@gmail.com>

ankush committed:
chore: simplify bug report form

nextchamp-saqib committed:
fix: set resolution by only if not on hold (#28995)

KrithiRamani committed:
resolved conflicts on merging with develop

KrithiRamani committed:
modularised create_delivery_note

KrithiRamani commented:
Made changes as suggested and reopening PR

KrithiRamani commented:
shows too many commits . will rebase and create new PR
